.github/workflows/pytest-gpu.yml

```diff
@@ -86,43 +35,70 @@ jobs:
       OMPI_CC: ${{ matrix.arch }}
 
     strategy:
-      # Prevent all build to stop if a single one fails
+      # Prevent all build from terminating if one fails
       fail-fast: false
 
       matrix:
         name: [
-          pytest-gpu-omp,
+          # NOTE: We can re-instate this as a 'failing' build
+          # as soon as the hardware is ready
+          #pytest-gpu-omp,
           pytest-gpu-acc
+          # pytest-gpu-aomp
         ]
         include:
-        - name: pytest-gpu-omp
-          test_file: "tests/test_gpu_openmp.py"
-          arch: "clang"
-          platform: "nvidiaX"
-          language: "openmp"
-          tags: ["self-hosted", "gpu", "openmp"]
+        #- name: pytest-gpu-omp
+          #test_file: "tests/test_gpu_openmp.py"
+          #env_file: "devito-ci-nvidia-openmp.env"
+          #test_drive_cmd: "nvidia-smi"
+          #arch: "clang"
+          #platform: "nvidiaX"
+          #language: "openmp"
+          #tags: ["self-hosted", "gpu", "openmp"]
 
         - name: pytest-gpu-acc
           test_file: "tests/test_gpu_openacc.py"
+          env_file: "devito-ci-nvidia-openacc.env"
+          test_drive_cmd: "nvidia-smi"
           arch: "pgcc"
           platform: "nvidiaX"
           language: "openacc"
           tags: ["self-hosted", "gpu", "openacc"]
 
+        # - name: pytest-gpu-aomp
+        #   test_file: "tests_gpu_aomp.py"
+        #   env_file: "devito-ci-amd-openmp.py"
+        #   test_drive_cmd: "TODO"
+        #   arch: "TODO"
+        #   platform: "TODO"
+        #   language: "TODO"
+        #   tags: ["self-hosted", "gpu", "aomp"]
+
     steps:
     - name: Checkout devito
       uses: actions/checkout@v1
 
+    # TODO: Below needs fixing. Suggest creating suitable matrix
+    # entries to load appropriate paths etc. from local files.
+    # See asv.yml for an example. Note that there may now be a nice action
+    # available for doing this so that would be worth looking into.
+    - name: Set environment
+      run: |
+        source $HOME/${{ matrix.env_file }}
+        echo "PATH=$PATH" >> $GITHUB_ENV
+        echo "LD_LIBRARY_PATH=$LD_LIBRARY_PATH" >> $GITHUB_ENV
+
     - name: Install dependencies
       run: |
         pip3 install --upgrade pip
         pip3 install -e .[extras]
 
     - name: Test with pytest
       run: |
-        if [ "${{ matrix.name }}" == 'pytest-gpu-acc' ]; then
-          pgaccelinfo
-        fi
+        # NOTE: Maybe we want to change the below to print
+        # output of nvidia-smi and amd equivalent on each machine?
```

Comment:
I will ask AMD what the equivalent is

Reply:
They suggest `/opt/rocm/bin/rocm_smi.py`

Reply:
@mloubout now that I think about it, that python wrapper for `nvidia-smi` maybe is already available somewhere in one the NVidia SDK subdirs.

Reply:
Left `TODO` since I didn't find `/opt/rocm/bin/rocm_smi.py`

Reply:
OK